packages/perseus-core/src/utils/split-perseus-item.ts

```diff
@@ -0,0 +1,36 @@
+import _ from "underscore";
+
+import {getPublicWidgetOptionsFunction} from "../widgets/core-widget-registry";
+
+import type {PerseusRenderer} from "../data-schema";
+
+export default function splitPerseusItem(
+    originalItem: PerseusRenderer,
+): PerseusRenderer {
+    const item = _.clone(originalItem);
+
+    if (!item.widgets || Object.keys(item.widgets).length === 0) {
```

Comment:
I forgot to upgrade widgets which I probably need to do first?

Reply:
Is this early return really needed? I might prefer to do something like

```
const upgradedWidgets = getUpgradedWidgetOptions(item.widgets ?? {});
```

which I think will be equivalent.

[Returning early just because data is empty can lead to bugs](https://bensguide.substack.com/i/150330330/consequential).

Reply:
I'm not sure I fully agree with you. One subtle difference:

``` TS
// my code
splitPerseusItem({}) // {}

// your code
splitPerseusItem({}) // { widgets: [] }
```

To me this feels wrong, but I guess so does continuing on.

Maybe the correct thing would be to throw and/or report the issue, but I think it's highly unlikely we hit this case anyway.